src/compile/data/index.ts

```diff
@@ -43,7 +44,28 @@ export interface DataComponent {
   isFaceted: boolean;
 
   /**
-   * Parse properties passed down from ancestors.
+   * Parse properties passed down from ancestors. Helps us to keep track of what has been parsed or is derived.
    */
-  ancestorParse: Dict<string>;
+  ancestorParse?: AncestorParse;
+}
+
+/**
+ * Class to track interesting properties (see https://15721.courses.cs.cmu.edu/spring2016/papers/graefe-ieee1995.pdf)
```

Comment:
Is this paper really needed?  It's another 8 page people have to read. 

Better provide a TLDR explanation.

Reply:
This goes into the depths of query optimization.